feat(dashboard): UI for charts dashboard page

mitchfaulcon opened the pull request:
### Purpose

<!---
Describe the problem or feature.
Consider providing an overview of why the work is taking place (with any relevant links); don’t assume familiarity with the history.
-->
Implemented the frontend for the charts page of the dashboard

<!---
Link to the related issues [KEY-000](link)
e.g [HRT-27](https://softeng761team5.atlassian.net/browse/HRT-27)
--->

[HRT-74](https://softeng761team5.atlassian.net/browse/HRT-74)

<!-- []() -->
<!-- []() -->

### Approach

<!---
How does this change address the problem
--->
- Added a similar page to the main dashboard page to view the charts for a specific course
- Pages are accessed via /Course/{courseID}
- Added more dummy data for temporary use, currently only the first course has associated charts
- Charts can be selected and the selected charts are stored in an array so that they can later be compared

***FYI: The Compare Charts text is currently off the button but will look fine once #37  is merged in***

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!--
If this PR contains a breaking change, describe the impact and migration path for existing applications below.
-->

### Screenshots before and after this change (required for UI changes)

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

<!---
Required if there is UI change
--->
n/a
#### Screenshot(s) of UI after this PR

<!---
Required if there is UI change
--->
![image](https://user-images.githubusercontent.com/42291691/94779171-ffaf2680-0422-11eb-8e84-79ac3eea2099.png)

mitchfaulcon committed:
feat(dashboard): implemented UI for charts page of dashboard

gargi-bansal reviewed (changes requested):
Mostly looks good to me, just a few UI changes.
- In one of my PRs I changed the styling of the plus image because it's a bit too big (I think in the preview PR). Probably should make it consistent.
- The checkboxes should be a different colour so that they're a bit more visible and obvious.
- The compare button could probably be disabled until at least 2 charts are selected.

gargi-bansal commented on src/components/Chart.tsx:
```diff
@@ -0,0 +1,47 @@
+}
+
+const Chart: React.FC<ChartProps> = (props: ChartProps) => {
+  const [chartProps, setChartProps] = useState(props);
```
is it necessary to store chart props? I assume the only thing you really need isSelected as state? The rest can just be accessible through props

mitchfaulcon commented:
> * In one of my PRs I changed the styling of the plus image because it's a bit too big (I think in the preview PR). Probably should make it consistent.

Yeah I saw that, I updated the classname for the new one I added so it should look fine once both our PR's are merged in

gargi-bansal commented on src/components/Course.tsx:
```diff
@@ -1,15 +1,19 @@
   const onClick = () => {
-    // TODO: Redirect to appropriate course page
+    history.push("/Course/" + props.courseID);
   };
```
Not necessary to change but you can also use backticks to put variables in a string.
E.g. history.push(`/Course/${props.courseID}`)

mitchfaulcon committed:
feat(charts dashboard): pR Comment fixes

mitchfaulcon committed:
refactor(charts): removed unnecessary prop for isSelected & changed compare button text

dder725 reviewed (approved):
LGTM!
